### ISSUE TITLE
invalid flag: --start-interval

### ISSUE THREAD

**Issue by MarcHagen** [hadolint/hadolint#978]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Linting file passes

### Actual behavior
/dev/stdin:4:52 invalid flag: --start-interval

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1
FROM scratch
HEALTHCHECK --start-period=10s --start-interval
```

**Additional environment details (OS, stack version, etc.)**
https://github.com/moby/moby/issues/45897
https://github.com/moby/moby/pull/40894
https://github.com/moby/buildkit/pull/3998
https://github.com/docker/cli/issues/4486

**Comment by felipecrs** [comment]
Guys, according to my tests, this flag does not do anything (it does not work). Is it working for any of you?

https://github.com/moby/buildkit/issues/4355

**Comment by kx1t** [comment]
It is a 

> Guys, according to my tests, this flag does not do anything (it does not work). Is it working for any of you?
> 
> [moby/buildkit#4355](https://github.com/moby/buildkit/issues/4355)

It *is* a valid option and hadolint should die over it.

**Comment by felipecrs** [comment]
Surely.

**Comment by tokohone** [comment]
This is indeed pretty irritating. So far no way to suppress this I guess.
Anybody looking at this currently? Now this option (i.e. "--start-interval=5s") can not be used in Dockerfile if hadolint is used.

**Comment by make-github-pseudonymous-again** [comment]
How can I ignore the rule that produces this error?

**Comment by make-github-pseudonymous-again** [comment]
> Second this bug / feature request as URGENT, especially since one cannot tell Hadolint to “ignore” it as it’s not reported as rule violation
> [docker/cli#4486](https://github.com/docker/cli/issues/4486)

I guess what @kx1t means is that [this error emerges from the parser itself](https://github.com/hadolint/language-docker/blob/dba56a780a50beb44c727a717db644b3da27320b/src/Language/Docker/Parser/Prelude.hs#L85). So, it is not implemented as a rule and cannot be disabled.

Should we create an feature request for this directly at https://github.com/hadolint/language-docker?

**Comment by thnee** [comment]
For reference, here is issue: https://github.com/hadolint/language-docker/issues/97, and pull request: https://github.com/hadolint/language-docker/pull/98

**Comment by make-github-pseudonymous-again** [comment]
> This is indeed pretty irritating. So far no way to suppress this I guess. Anybody looking at this currently? Now this option (i.e. "--start-interval=5s") can not be used in Dockerfile if hadolint is used.

> How can I ignore the rule that produces this error?

```
grep -v start-interval Dockerfile | sudo docker run --rm -i hadolint/hadolint
```

HAHA!

**Comment by 304NotModified** [comment]
As https://github.com/hadolint/language-docker/pull/98 had been merged and there is a release of hadolint after it, this is resolved?

**Comment by m-ildefons** [comment]
Closing this issue. `--start-interval` flag is supported from Hadolint v2.14.0 onwards.
